netbox/core/jobs.py

```diff
@@ -31,7 +31,6 @@ def enqueue(cls, *args, **kwargs):
 
         # Update the DataSource's synchronization status to queued
         if datasource := job.object:
```

Comment:
You've removed the set to datasource.status but it is still being used in the update clause which means it will get set to whatever job.object currently has. I think we want to set it to a set value here?